File: 2025 APRIL(LEETCODE)/DAY 1/DAY_1.java

class DAY_1 {
    public static void main(String[] args) {
        // Example usage
        int[][] questions = {{1, 2}, {2, 3}, {3, 4}};
        System.out.println(solve(questions)); // Output: 4
    }

    public static long solve(int[][] questions) {
        int n = questions.length;
        long[] dp = new long[n + 1];
        dp[0] = 0;

        for (int i = 1; i <= n; i++) {
            dp[i] = Math.max(dp[i - 1], dp[i - 1] + questions[i - 1][0]);
            if (i > questions[i - 1][1]) {
                dp[i] = Math.max(dp[i], dp[i - questions[i - 1][1] - 1] + questions[i - 1][0]);
            }
        }

        return dp[n];
    }
}

File: 2025 APRIL(LEETCODE)/DAY 1/DAY_1.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

long solve(vector<vector<int>>& questions) {
    int n = questions.size();
    vector<long> dp(n + 1, 0);

    for (int i = n - 1; i >= 0; i--) {
        int points = questions[i][0];
        int brainpower = questions[i][1];
        int nextIndex = i + brainpower + 1;

        long skip = dp[i + 1];
        long solve = points + (nextIndex < n ? dp[nextIndex] : 0);

        dp[i] = max(skip, solve);
    }

    return dp[0];
}

int main() {
    // Example usage
    vector<vector<int>> questions = {
        {3, 2},
        {4, 3},
        {4, 4},
        {2, 5}
    };
    cout << solve(questions) << endl; // Output: 5
    return 0;
}